### PR TITLE
Escape \ as well as " in message property

### LINKED ISSUE #117: PullRequest decoration (Github) fails due to error content
**Describe the bug**
On Github, pull request decoration fails on an GraphQL API call apparently due to the error to report content.

**To Reproduce**
Have SonarQube reporting an error with a message that contains special characters, like:
`Rename this local variable to match the regular expression "^(@{0,2}[\da-z_]+[!?=]?)|([*+-/%=!><~]+)|(\[]=?)$".`

Pull request decoration fails with the following error
```
java.lang.IllegalStateException: Could not decorate Pull Request on Github
        at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.github.GithubPullRequestDecorator.decorateQualityGateStatus(GithubPullRequestDecorator.java:38)
        at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.PullRequestPostAnalysisTask.finished(PullRequestPostAnalysisTask.java:132)
        at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.executeTask(PostProjectAnalysisTasksExecutor.java:113)
        at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.finished(PostProjectAnalysisTasksExecutor.java:107)
        at org.sonar.ce.task.step.ComputationStepExecutor.executeListener(ComputationStepExecutor.java:91)
        at org.sonar.ce.task.step.ComputationStepExecutor.execute(ComputationStepExecutor.java:63)
        at org.sonar.ce.task.projectanalysis.taskprocessor.ReportTaskProcessor.process(ReportTaskProcessor.java:81)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.executeTask(CeWorkerImpl.java:209)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.run(CeWorkerImpl.java:191)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.findAndProcessTask(CeWorkerImpl.java:158)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$TrackRunningState.get(CeWorkerImpl.java:133)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:85)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:53)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: An error was returned in the response from the Github API:
- Error{message='Parse error on "\"" (error) at [1, 1217]', locations=[Location{line='1', column='1217'}]}
        at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.github.v4.GraphqlCheckRunProvider.createCheckRun(GraphqlCheckRunProvider.java:173)
        at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.github.GithubPullRequestDecorator.decorateQualityGateStatus(GithubPullRequestDecorator.java:36)
        ... 19 common frames omitted
```

**Expected behavior**
Pull request decoration to succeed with the error being reported

**Software Versions**
 - SonarQube Version: Version 7.9.1 (build 27448)
 - Plugin Version: 1.3.1-SNAPSHOT

### PR DESCRIPTION
**Pull request by hnioche**
This fixes #117

### CODE REVIEW

**Comment by marvin-w** [comment]
Can you add a test for it please so it doesn't happen again in the future? :-)

**Comment by hnioche** [comment]
> Can you add a test for it please so it doesn't happen again in the future? :-)

Will add this as soon as I have a few minutes

**Comment by hnioche** [comment]
I've added a unit test. I'm not used to work with Java and its environment so I'm not sure I've done this the best way possible.

File: src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/github/v4/GraphqlCheckRunProviderTest.java [hunk 1]
```diff
@@ -195,12 +195,33 @@ public void createCheckRunHappyPathErrorStatus() throws IOException, GeneralSecu
...
+            "issue 1",
+            "issue 2",
+            "issue 3",
+            "issue 4",
+            "issue 5",
+            "issue 6",
+            "issue \\ \" \\\" \"\\ \\\\ \"\" 7"
+        };
+
+        String[] messageOutput ={
```
Comment (mc1arke) [inline src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/github/v4/GraphqlCheckRunProviderTest.java]: Formatting: could you ad a space between the `=` and the `{` please?

Comment (hnioche) [reply src/test/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/github/v4/GraphqlCheckRunProviderTest.java]: This is fixed now

**Comment by mc1arke** [comment]
Please squash your commits so there's a single commit to cover your feature

**Comment by mc1arke** [comment]
Merged. Thanks for the contribution

**Pull request merged**